src/main/java/storagesystem/controller/LoginPageController.java

```diff
@@ -0,0 +1,183 @@
+
+        assignHandlers();
+    }
+
+    private void assignHandlers() {
+        ArrayList<Control> loginFields = new ArrayList();
+        loginFields.add(userNameTextField);
+        loginFields.add(passwordTextField);
+        loginFields.add(loginButton);
+        loginFields.add(organisationChoiceBox);
+
+        for (Control loginField :
+                loginFields) {
+            loginField.setOnKeyPressed(event -> {
+                if (event.getCode() == KeyCode.ENTER) {
+                    try {
+                        loginButtonPressed();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * attempts to login with the entered credentials in userNameTextField and passwordTextField
+     */
+    @FXML
+    private void loginButtonPressed() throws IOException {
+        //check username and password against database
+        if (doesUserExist()) {
+            //set current user
+            StorageSystem.setCurrentUser(loginUser);
+            StorageSystem.setCurrentOrganisation(getSelectedOrganisation());
+
+            //open dashboard
+            Parent root = FXMLLoader.load(getClass().getResource("/dashboard.fxml"));
+            Stage stage = (Stage) rootPane.getScene().getWindow();
+            stage.setScene(new Scene(root));
+            stage.show();
+        } else {
+            System.out.println("User with name \"" + userNameTextField.getText() + "\" does not exist");
+        }
+    }
+
+    @FXML
+    private void registerButtonPressed() throws IOException {
+        //make sure there is no user with the name
+        if (!doesUserExist()) {
+            String name = userNameTextField.getText();
+            getSelectedOrganisation().createUser(name);
+            fadeTransition(userRegisteredLabel);
+        } else {
+            System.out.println("A user with that name already exists");
+        }
+    }
+
+    /**
+     * Check if selected value in the Organisation Choicebox actually corresponds to an existing organisation in the database
+     *
+     * @return The actual organisation from the database
+     */
+    private Organisation getSelectedOrganisation() {
+        String selectedOrganisation = organisationChoiceBox.getValue().toString();
+        for (Organisation org :
+                StorageSystem.getOrganisations()) {
+            if (org.getName().equals(selectedOrganisation)) {
+                return org;
+            }
+        }
+        //todo throw exception? or write error message saying the admin needs to add organisations
+        return null;
+    }
+
+    /**
+     * Checks in the selected organisation if there is an user with the name currently written in the Username textfield
+     *
+     * @return
+     */
+    private boolean doesUserExist() {
+        Organisation selectedOrganisation = getSelectedOrganisation();
```

Comment:
getSelectedOrganisation() will check the String value in the Choicebox and then cross reference it to the model and see if there is any organisation with that name.

Will not ALWAYS return null if we haven't actively selected something since the program chooses first value in the Choicebox by default.